### PR TITLE
[Improvement-#11863][task-plugin] add the function of stopping SQL type tasks

### PR DESCRIPTION
**Pull request by fengjian1129**
## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

**Comment by fengjian1129** [comment]
@ruanwenjun hi bro,review the code,Thanks

**Comment by fengjian1129** [comment]
@caishunfeng hi bro,review the code,Thanks

File: dolphinscheduler-task-plugin/dolphinscheduler-task-sql/pom.xml [hunk 1]
```diff
@@ -44,6 +44,12 @@
             <artifactId>dolphinscheduler-datasource-all</artifactId>
             <version>${project.version}</version>
         </dependency>
+
+        <dependency>
+            <groupId>org.apache.dolphinscheduler</groupId>
+            <artifactId>dolphinscheduler-server</artifactId>
+            <version>${project.version}</version>
+        </dependency>
```
Comment (ruanwenjun) [inline dolphinscheduler-task-plugin/dolphinscheduler-task-sql/pom.xml]: please remove this.

File: dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/main/java/org/apache/dolphinscheduler/plugin/task/sql/SqlTask.java [hunk 2]
```diff
@@ -167,7 +173,13 @@ public void handle(TaskCallBack taskCallBack) throws TaskException {
 
     @Override
     public void cancel() throws TaskException {
-
+        String type = sqlParameters.getType();
+        if (DbType.HIVE == DbType.valueOf(type)) {
+            List<String> appIds = ProcessUtils.killYarnJob(taskExecutionContext);
```
Comment (ruanwenjun) [inline dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/main/java/org/apache/dolphinscheduler/plugin/task/sql/SqlTask.java]: Can we move this in `org.apache.dolphinscheduler.plugin.task.api.ProcessUtils` 
cc @caishunfeng

**Review by ruanwenjun — CHANGES_REQUESTED** [review]

File: dolphinscheduler-task-plugin/dolphinscheduler-task-sql/pom.xml (hunk 1 above)
Comment (caishunfeng) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-sql/pom.xml]: +1

File: dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/main/java/org/apache/dolphinscheduler/plugin/task/sql/SqlTask.java (hunk 2 above)
Comment (caishunfeng) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/main/java/org/apache/dolphinscheduler/plugin/task/sql/SqlTask.java]: > Can we move this in `org.apache.dolphinscheduler.plugin.task.api.ProcessUtils` cc @caishunfeng

Yes, agree with wenjun, we shouldn't maintain multiple copies of the same code

Comment (fengjian1129) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/main/java/org/apache/dolphinscheduler/plugin/task/sql/SqlTask.java]: Can I write a method of kill yarn's job in the [task-api] ? No available tool classes found in [task-plugin]

**Comment by fengjian1129** [comment]
@ruanwenjun @caishunfeng I have modified the code and many tools classes have been merged, and the local test has been completed

**Comment by fengjian1129** [comment]
@caishunfeng UT failed.... I haven't modified this method(org.apache.dolphinscheduler.server.utils. ProcessUtilsTest .getPidsStr())

**Pull request closed without merging**